flask_mongorest/resources.py

```diff
@@ -199,15 +201,26 @@ def serialize_field(self, obj, **kwargs):
         else:
             return self.serialize(obj, **kwargs)
 
+    def _subresource(self, obj):
+        """Selects and creates an appropriate sub-resource class for delegation or return None if there isn't one"""
+        default_s_class = self._child_document_resources[self.default_child_resource_document] if self.default_child_resource_document else None
+        s_class = self._child_document_resources.get(obj.__class__, default_s_class)
```

Comment:
Assuming that `_child_document_resources[obj.__class__]` never returns a falsey value, the code below is equivalent and IMHO much cleaner. Do you agree @tsx?

```
s_class = self._child_document_resources.get(obj.__class__)
if not s_class and self.default_child_resource_document:
    s_class = self._child_document_resources[self.default_child_resource_document]
```